src/preferences/common/preference-service.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { Emitter, Event } from "../../application/common";
+import { inject, injectable } from "inversify";
+import URI from "../../application/common/uri";
+import { FileSystem, FileChangesEvent } from "../../filesystem/common/filesystem";
+import { FileSystemWatcher } from "../../filesystem/common/filesystem-watcher"
+import * as coreutils from "@phosphor/coreutils";
+
+
+export interface PreferenceChangedEvent {
+    readonly preferenceName: string;
+    readonly newValue?: any;
+    readonly oldValue?: any;
+}
+
+export const PreferenceService = Symbol("PreferenceService")
+export const PreferencePath = Symbol("PreferencePath")
+
+
+/**
+ * Minimal preference API with basic functionalities
+ */
+export interface PreferenceService {
+    // A minimal part of the interface.
+    has(preferenceName: string): Promise<boolean>;
+    get<T>(preferenceName: string): Promise<T | undefined>;
+    readonly onPreferenceChanged: Event<PreferenceChangedEvent>;
```

Comment:
I don't think it is going to work with JSON-RPC, `PreferenceServiceClient` should be used to listen for changes